### ISSUE TITLE
Title: Conflict Between language='en' and forced_decoder_ids in Whisper Model & Attention Mask Warning

### ISSUE THREAD

**Issue by priyam-hub** [priyam-hub/Speechify#1]
Description:
I encountered warnings and potential unexpected behavior while transcribing audio using the Whisper multilingual model from Hugging Face Transformers. The issues are:

### 1️⃣ Conflict Between `language='en'` and `forced_decoder_ids`

- When setting `language='en'`, the following warning appears:

```pgsql
You have passed language=en, but also have set forced_decoder_ids to [[1, None], [2, 50359]] which creates a conflict.
forced_decoder_ids will be ignored in favor of language=en.
```
### 2️⃣ Attention Mask Warning
- When calling `.generate()` on processed input, the following warning appears:
```vbnet
The attention mask is not set and cannot be inferred from input because pad token is same as eos token.
As a consequence, you may observe unexpected behavior.
Please pass your input's attention_mask to obtain reliable results.
```

- The documentation does not specify how to properly handle this situation, leading to confusion.

### Expected Behavior:

- If `language='en'` is explicitly set, `forced_decoder_ids` should either be automatically adjusted without a warning or a clearer error should be provided.

- If `attention_mask` is missing, there should be a default fallback mechanism instead of just a warning.

### Steps to Reproduce:

```python
from transformers import WhisperProcessor, WhisperForConditionalGeneration

# Load model and processor
processor = WhisperProcessor.from_pretrained("openai/whisper-large-v3")
model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-large-v3")

# Dummy audio input
import numpy as np
audio_input = np.random.rand(16000)  # Simulated 1-second audio

# Process input
input_features = processor(audio_input, sampling_rate=16000, return_tensors="pt")

# Generate transcription
predicted_ids = model.generate(input_features.input_features, language='en')

# Decode output
transcription = processor.batch_decode(predicted_ids, skip_special_tokens=True)[0]

print(transcription)
```
